lib/ui/form/form_view_model.dart

```diff
@@ -27,10 +37,45 @@ class FormViewModel extends StateNotifier<FormState> {
     );
     if (result is Success<SurveyDetailsModel>) {
       _surveyDetails.add(result.value);
+      _outroMessage = result.value.outro;
       state = const FormState.loadSurveyDetailsSuccess();
     } else {
       _error.add((result as Failed).getErrorMessage());
       state = const FormState.loadSurveyDetailsError();
     }
   }
+
+  void saveIndexedAnswer(
+    String questionId,
+    List<SubmitSurveyAnswerModel> answers,
+  ) {
+    final question = submitSurveyQuestions
+        .firstWhereOrNull((question) => question.id == questionId);
+
+    if (question == null) {
+      submitSurveyQuestions.add(SubmitSurveyQuestionModel(
+        id: questionId,
+        answers: answers,
+      ));
+    } else {
+      question.answers.clear();
+      question.answers.addAll(answers);
+    }
+  }
+
+  void submitAnswer(String surveyId) async {
+    state = const FormState.loading();
+    final result = await _submitSurveyUseCase.call(
+      SubmitSurveyInput(
+        surveyId: surveyId,
+        questions: submitSurveyQuestions,
+      ),
+    );
+    if (result is Success<void>) {
+      state = FormState.surveySubmissionSuccess(_outroMessage ?? '');
```

Comment:
Can't we get this message from `_surveyDetails` instead of from an additional cache local val?